src/components/big-interactive-pages/editor.tsx

```diff
@@ -145,6 +163,38 @@ export default function Editor({ persistenceState, cookies }: EditorProps) {
 	const screenContainer = useRef<HTMLDivElement>(null);
 	const screenControls = useRef<HTMLDivElement>(null);
 
+	const [sessionId] = useState(nanoid());
+
+	useEffect(() => {
```

Comment:
It is true that all this does detect new sessions?

Reply:
yes currently all it does is communicate with session on each tabs aka broadcasting the message to all tabs. (mostly used for testing purposes looking at it now)

Reply:
can this be scoped out and separated to make the logic of how the editor is init'd more concise?

Reply:
i.e. useEffect(logNewSessions)